tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -150,8 +150,7 @@ class SourceMapWriter(
     pendingColumnInGenerated = startColumn
     pendingPos = originalPos
     pendingName =
-      if (startColumn != pendingColumnInGenerated) originalName
-      else if (pendingName != null) pendingName
+      if (pendingName != null) pendingName
       else originalName
```

Comment:
Actually, it can go further and just do
```scala
pendingName = originalName
```
The previous logic was completely broken. I have no idea how I came to write this stupidity back then. Possible explanation: I wrote that 3 days before Scala Days 2013, where I was going to present Scala.js to the world for the first time, and I wanted source maps to get in ^^